### PR TITLE
Bump pull-boskos-build-test-verify to use golang 1.22.5

### PR DESCRIPTION
**Pull request by dims**
Needed for bumping versions in https://github.com/kubernetes-sigs/boskos/pull/199

### CODE REVIEW

File: config/jobs/kubernetes-sigs/boskos/boskos-presubmits.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ presubmits:
       path_alias: sigs.k8s.io/boskos
       spec:
         containers:
-          - image: public.ecr.aws/docker/library/golang:1.22.3
+          - image: public.ecr.aws/docker/library/golang:1.22.5
```
Comment (BenTheElder) [inline config/jobs/kubernetes-sigs/boskos/boskos-presubmits.yaml:10]: TODO: switch to go toolchain?

for a project like boskos we can probably trivially adopt the go.mod statement?

Comment (dims) [reply config/jobs/kubernetes-sigs/boskos/boskos-presubmits.yaml:10]: ACK i see a lot more stuff to be done in boskos as well :(

**Pull request merged**